src/cmds/env.rs

```diff
@@ -113,14 +113,35 @@ pub fn generate_keypair(args: GenerateKeypair) -> Result<Msg> {
         },
         &mut OsRng,
     );
-    if let Some(parent) = args.out_file.parent() {
-        fs::create_dir_all(parent)?;
+
+    let do_write = |msg: String| -> Result<Msg> {
+        if let Some(parent) = args.out_file.parent() {
```

Comment:
Shouldn’t this be `if let None` why would you need to create all the directories in the supplied path if it already exists?